docs/dev.md

```diff
@@ -0,0 +1,420 @@
+# Development
+
+You can build, check, and test dcos-net in a development image using
+`make docker-compile`, `make docker-check`, and `make docker-test` respectively.
+All makefile targets with `docker-` prefix build development image with all
+dependencies and run `rebar3` in that image on the host directory.
+
+To check your dcos-net build on DC/OS you can use DC/OS E2E a.k.a.
+[dcos-docker CLI](http://dcos-e2e.readthedocs.io/en/latest/cli.html):
+
+```sh
+curl -O https://downloads.dcos.io/dcos/testing/master/dcos_generate_config.sh
+make dcos-docker-create DCOS_DOCKER_AGENTS=3
+make dcos-docker-dev DCOS_DOCKER_NODE=master_0
+make dcos-docker-shell DCOS_DOCKER_NODE=agent_1
+make dcos-docker-destroy
+```
```

Comment:
What do you think about making the sessions clearly separated? between the simple way to do it and the advanced.

```md
## Basic Development
- curl -O https://downloads.dcos.io/dcos/testing/master/dcos_generate_config.sh
- make dcos-docker-create DCOS_DOCKER_AGENTS=3


## Advanced Development

Alternatively....
```